### PR TITLE
Added s3 plugin to Es5.x to enable s3 backups

### PR DESCRIPTION
**Pull request by shgriffi**
Added s3 plugin in order to enable backups and restores of indexes to S3.

### CODE REVIEW

File: elasticsearch/prep-install.origin [hunk 1]
```diff
@@ -8,4 +8,7 @@ fi
 if [ -z "${PROMETHEUS_EXPORTER_URL:-}" ] ; then
     PROMETHEUS_EXPORTER_URL=https://distfiles.compuscene.net/elasticsearch/elasticsearch-prometheus-exporter-${PROMETHEUS_EXPORTER_VER}.zip
 fi
-es_plugins=($ES_CLOUD_K8S_URL $OSE_ES_URL $PROMETHEUS_EXPORTER_URL)
+if [ -z "${ADDITIONAL_PLUGINS:-}" ] ; then
+    ADDITIONAL_PLUGINS=
+fi
+es_plugins=($ES_CLOUD_K8S_URL $OSE_ES_URL $PROMETHEUS_EXPORTER_URL $ADDITIONAL_PLUGINS)
```
Comment (richm) [inline elasticsearch/prep-install.origin]: You don't need the `if` clause, you can just write it as one line:
`es_plugins=($ES_CLOUD_K8S_URL $OSE_ES_URL $PROMETHEUS_EXPORTER_URL ${ADDITIONAL_PLUGINS:-})`

**Comment by richm** [comment]
Do you need another Dockerfile?  Can you pass in the additional plugins to use like this, with your first commit?
```
docker build -e ADDITIONAL_PLUGINS=repository-s3 ....
```
or using `build_arg`:
```
docker build --build-arg=ADDITIONAL_PLUGINS=repository-s3 ....
```
?

File: elasticsearch/prep-install.origin [hunk 2]
```diff
@@ -8,4 +8,7 @@ fi
 if [ -z "${PROMETHEUS_EXPORTER_URL:-}" ] ; then
     PROMETHEUS_EXPORTER_URL=https://distfiles.compuscene.net/elasticsearch/elasticsearch-prometheus-exporter-${PROMETHEUS_EXPORTER_VER}.zip
 fi
-es_plugins=($ES_CLOUD_K8S_URL $OSE_ES_URL $PROMETHEUS_EXPORTER_URL)
+if [ -z "${ADDITIONAL_PLUGINS:-}" ] ; then
```
Comment (jcantrill) [inline elasticsearch/prep-install.origin]: In general, I like this addition to allow users to customize their own images.

File: elasticsearch/Dockerfile.centos7.s3repository [hunk 3]
```diff
@@ -0,0 +1,66 @@
+FROM centos:centos7
```
Comment (jcantrill) [inline elasticsearch/Dockerfile.centos7.s3repository]: @richm @ewolinetz In general, I don't like the idea of adding a whole other Dockerfile.  Shouldn't we be able to build upon one that exists by adding env vars and args to one which already exists?

Comment (richm) [reply elasticsearch/Dockerfile.centos7.s3repository]: > Shouldn't we be able to build upon one that exists by adding env vars and args to one which already exists?

Yes, we should

Comment (ewolinetz) [reply elasticsearch/Dockerfile.centos7.s3repository]: I agree, I feel that if we have another Dockerfile that implies we fully support and will provide images for that Dockerfile.

If we provide the means to support custom images where a customer can specify and build with additional plugins I would prefer to go that route for now.

**Comment by shgriffi** [comment]
Thanks for the feedback everyone.  Makes perfect sense to not introduce another docker file.  I'll rework this to take that into account.

**Pull request closed without merging**